### PR TITLE
Add prune command

### PR DESCRIPTION
**Pull request by zelahi**
This logic brings back the prune cmd and mimics the behavior over here: https://github.com/docker/cli/blob/2e385015f73157af8673263a9e7ccb16ad04fad1/cli/command/builder/prune.go

It references: #55

### CODE REVIEW

**Comment by zelahi** [comment]
It was interesting as it failed when I auto-triggered the job.  However it looked like it passed once I logged in to Travis Ci and ran the job

File: commands/prune.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	warning := normalWarning
+	if options.all {
+		warning = allCacheWarning
+	}
+	if !options.force && !command.PromptForConfirmation(dockerCli.In(), dockerCli.Out(), warning) {
+		return 0, "", nil
+	}
+
+	report, err := dockerCli.Client().BuildCachePrune(context.Background(), types.BuildCachePruneOptions{
```
Comment (tonistiigi) [inline commands/prune.go]: this needs to call to the buildkit grpc client Prune, not to docker cli. https://github.com/moby/buildkit/blob/master/client/prune.go

Comment (zelahi) [reply commands/prune.go]: gotcha thanks!  I'll make that change

**Comment by zelahi** [comment]
@tonistiigi I made the changes requested using the moby client api for prune instead of the docker cli. 
 One other thing I noticed is that this line here changes the way `NewAuthProvider` is being used: https://github.com/docker/buildx/blob/e3a251e962cb7aaba0a91db9f569559bb089e504/vendor/github.com/moby/buildkit/session/auth/authprovider/authprovider.go#L15-L19

Which caused changes to happen in `bake.go` and `build.go` and arrived here in this commit

**Comment by AkihiroSuda** [comment]
needs rebase

**Comment by tonistiigi** [comment]
superseded by #249

**Pull request closed without merging**